Xamarin.Essentials/FilePicker/FilePicker.shared.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Xamarin.Essentials
+{
+    public static partial class FilePicker
+    {
+        public static Task<PickResult> PickFileAsync(PickOptions options = null)
+            => PlatformPickFileAsync(options ?? PickOptions.Default);
+    }
+
+    public partial class FilePickerFileType
+    {
+        public static readonly FilePickerFileType Images = PlatformImageFileType();
+        public static readonly FilePickerFileType Png = PlatformPngFileType();
+
+        readonly IDictionary<DevicePlatform, IEnumerable<string>> fileTypes;
+
+        protected FilePickerFileType()
+        {
+        }
+
+        public FilePickerFileType(IDictionary<DevicePlatform, IEnumerable<string>> fileTypes) => this.fileTypes = fileTypes;
+
+        public IEnumerable<string> Value => GetPlatformFileType(DeviceInfo.Platform);
+
+        protected virtual IEnumerable<string> GetPlatformFileType(DevicePlatform platform)
+        {
+            if (fileTypes.TryGetValue(platform, out var type))
+                return type;
+
+            throw new PlatformNotSupportedException("This platform does not support this file type.");
+        }
+    }
+
+    public class PickOptions
+    {
+        public static PickOptions Default
+            => new PickOptions
+            {
+                FileTypes = null,
+            };
+
+        public static PickOptions Images
+            => new PickOptions
+            {
+                FileTypes = FilePickerFileType.Images
+            };
+
+        public string PickerTitle { get; set; }
+
+        public FilePickerFileType FileTypes { get; set; }
+    }
+
+    public partial class PickResult
```

Comment:
This could probably inherit from `FileBase`... Then the bits for path/uri may not need duplication. Not sure if it actually will fit. @Redth we probably need to see if we did that right for the "picker" concept. I see I did something similar way back when: https://github.com/xamarin/Essentials/pull/613/files#diff-160cde01258a9111337b1e0b33916789R20

Reply:
It depends on if the FileBase capsulates a real file. In file picking, at least on Android, the returned content URI is almost never a file, and the assumption that a file is being returned from picking confused and distracted quite a few users.